src/app/components/user-details/user-details.component.html

```diff
@@ -0,0 +1,46 @@
+<div class="user-details">
+    <form [formGroup]="userFormGroup">
+        <div>
+            <label for="username">Username:</label><br>
+            <input type="text" formControlName="username">
+        </div>
+        <div>
+            <label for="first_name">First name:</label><br>
+            <input type="text" formControlName="first_name">
+        </div>
+        <div>
+            <label for="last_name">Last name:</label><br>
+            <input type="text" formControlName="last_name">
+        </div>
+        <div>
+            <label for="Roles">
+                Roles: <br>
+                    @for (role of user?.roles; track role) {
+                    <ng-container formArrayName="roles">
```

Comment:
this is a little confusing you're iterating through roles (why is user as an optional field? a user cannot go to user-details page without any user object) and then inside the for loop you have a formArray which will ? it's confusing :D